sdk/eventhub/event-hubs/test/eventProcessor.spec.ts

```diff
@@ -484,5 +484,114 @@ describe("Event Processor", function(): void {
       partitionOwnershipList[0].sequenceNumber!.should.equals(checkpoint.sequenceNumber);
       partitionOwnershipList[0].offset!.should.equals(checkpoint.offset);
     });
+
+    it("should receive events from the checkpoint", async function(): Promise<void> {
+      const partitionIds = await client.getPartitionIds();
+
+      // ensure we have at least 2 partitions
+      partitionIds.length.should.gte(2);
+
+      let partitionResultsMap = new Map<string, ReceivedEventData[]>();
+      partitionIds.forEach((id) => partitionResultsMap.set(id, []));
+      let didError = false;
+
+      // The partitionProcess will need to add events to the partitionResultsMap as they are received
+      const factory: PartitionProcessorFactory = (context, checkpointManager) => {
+        return {
+          async processEvents(events: ReceivedEventData[]) {
+            const existingEvents = partitionResultsMap.get(context.partitionId)!;
+            events.forEach((event: ReceivedEventData) => {
+              existingEvents.push(event);
+              debug(
+                "Received event: '%s' from partition: '%s' and consumer group: '%s'",
+                event.body,
+                context.partitionId
+              );
+              checkpointManager.updateCheckpoint(event);
+            });
+          },
+          async processError() {
+            didError = true;
+          }
+        };
+      };
+
+      const inMemoryPartitionManager = new InMemoryPartitionManager();
+      const processor1 = new EventProcessor(
+        EventHubClient.defaultConsumerGroupName,
+        client,
+        factory,
+        inMemoryPartitionManager,
+        {
+          initialEventPosition: EventPosition.fromEnqueuedTime(new Date())
+        }
+      );
+
+      // start first processor
+      processor1.start();
+
+      // create messages
+      const expectedMessagePrefix = "EventProcessor test - checkpoint - ";
+      const events: EventData[] = [];
+
+      for (const partitionId of partitionIds) {
+        const producer = client.createProducer({ partitionId });
+        for (let index = 1; index <= 100; index++) {
+          events.push({ body: `${expectedMessagePrefix} ${index} ${partitionId}` });
+        }
+        await producer.send(events);
+        await producer.close();
+      }
+
+      // set a delay to give a consumers a chance to receive a message
+      await delay(500);
```

Comment:
Might need a longer delay just based on what I saw with a previous flaky test. 3 seconds seemed to be a reliable value.